content/300-guides/100-performance-and-optimization/150-connection-management/200-configure-pg-bouncer.mdx

```diff
@@ -29,3 +29,17 @@ _Transaction mode_ offers a connection for every transaction – a requirement f
 - Prisma opens a transaction for every query case – even when just reading data, allowing Prisma to use prepared statements.
 
 </details>
+
+## Prisma Migrate and PgBouncer workaround
+
+Prisma Migrate uses **database transactions** to check out the current state of the database and the migrations table. However, the Migration Engine is designed to use a **single connection to the database**, and does not support connection pooling with PgBouncer. If you attempt to run Prisma Migrate commands in any environment that uses PgBouncer, you will see the following error:
+
+```bash
+Error: undefined: Database error
+Error querying the database: db error: ERROR: prepared statement "s0" already exists
+```
+
+To work around this issue, you must connect directly to the database rather than going through PgBouncer. How to achieve this depends on your setup or provider:
+
+* [Connecting directly to a PostgreSQL database hosted on Digital Ocean](https://github.com/prisma/prisma/issues/6157)
+* [Connecting directly to a PostgreSQL database hosted on ScaleGrid](https://github.com/prisma/prisma/issues/6701#issuecomment-824387959)
```

Comment:
This could probably be generalized - but as we are currently working on getting rid of this limitation, it might not be worth the effort.